<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mips</class>
 <widget class="QWidget" name="mips">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1239</width>
    <height>796</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>mips</string>
  </property>
  <property name="styleSheet">
   <string notr="true">*{
	font: 25 10pt &quot;微软雅黑 Light&quot;;
}

QWidget#mips{
	background-color: rgb(31, 36, 48);
}

QPushButton{
	background-color: rgb(31, 36, 48);
	color: white;
	padding-left: 10px;
	padding-right: 10px;
	padding-top: 4px;
	padding-bottom: 4px;
	width: 80px;
	margin: 0px 0xp 0px 0px;
	border-right: none;
	border-top: none;
	border-bottom: 1px ridge #3498db;
	border-left: none;
}

QPushButton:hover{
	color: rgb(115, 208, 255);
}

QPushButton:pressed{
	padding-left: 8px;
	padding-top: 2px;
}

QLabel{
	padding-top: 4px;
	padding-bottom: 4px;
}

QLineEdit {
background-color: rgb(35, 40, 52);
color: white;
	border-right: none;
	border-top: none;
	border-bottom: 1px ridge #3498db;
	border-left: none;
	padding-top: 3px;
	padding-bottom: 3px;
}</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QWidget" name="functionArea" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>40</height>
      </size>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="spacing">
       <number>9</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QPushButton" name="newFile">
        <property name="text">
         <string>新建文件</string>
        </property>
        <property name="shortcut">
         <string>Ctrl+N</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="openFile">
        <property name="text">
         <string>打开文件</string>
        </property>
        <property name="shortcut">
         <string>Ctrl+O</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="saveFile">
        <property name="text">
         <string>保存</string>
        </property>
        <property name="shortcut">
         <string>Ctrl+S</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="compile">
        <property name="text">
         <string>编译</string>
        </property>
        <property name="shortcut">
         <string>Ctrl+W</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="run">
        <property name="text">
         <string>运行</string>
        </property>
        <property name="shortcut">
         <string>Ctrl+R</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="debug">
        <property name="text">
         <string>调试</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="runOneLine">
        <property name="text">
         <string>单步运行</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="title">
        <property name="styleSheet">
         <string notr="true">font: 12pt &quot;华文隶书&quot;;
margin-top: 3px;
padding-right: 10px;
color: #ff7675;</string>
        </property>
        <property name="text">
         <string>Mips模拟器</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="mainArea" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="5,2">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QWidget" name="codeContainer" native="true">
        <property name="styleSheet">
         <string notr="true"/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="hexCodeContainer" native="true">
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QPlainTextEdit" name="regCode">
           <property name="styleSheet">
            <string notr="true">background-color: rgb(35, 40, 52);
color: rgb(203, 204, 198);
font: 12pt Consolas;
border: 1px solid #2d3436;
border-left: none;</string>
           </property>
           <property name="documentTitle">
            <string/>
           </property>
           <property name="undoRedoEnabled">
            <bool>false</bool>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="backgroundVisible">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <property name="spacing">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="label_2">
             <property name="styleSheet">
              <string notr="true">color: white;</string>
             </property>
             <property name="text">
              <string>将</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="regNumber">
             <property name="styleSheet">
              <string notr="true">width: 28px;</string>
             </property>
             <property name="maxLength">
              <number>2</number>
             </property>
             <property name="placeholderText">
              <string>0</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_3">
             <property name="styleSheet">
              <string notr="true">color: white;</string>
             </property>
             <property name="text">
              <string>寄存器设置为</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="number">
             <property name="styleSheet">
              <string notr="true"/>
             </property>
             <property name="maxLength">
              <number>10</number>
             </property>
             <property name="placeholderText">
              <string>0</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="editReg">
             <property name="text">
              <string>设置</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="consoleArea" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>230</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>230</height>
      </size>
     </property>
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QTabWidget" name="tabArea">
        <property name="styleSheet">
         <string notr="true">QTabWidget::pane{
    border: none;
}

QTabWidget::tab-bar {
    left: 5px;
}

QTabBar::tab {
	font: 25 9pt &quot;微软雅黑 Light&quot;;
	color: white;
	background-color: rgb(31, 36, 48);
	border: none;
	width: 60px;
	padding: 4px 7px;
}

QTabBar::tab:selected {
	border-bottom: 1px ridge #3498db;
}</string>
        </property>
        <property name="currentIndex">
         <number>1</number>
        </property>
        <widget class="QWidget" name="console">
         <property name="styleSheet">
          <string notr="true">background-color: rgb(31, 36, 48);</string>
         </property>
         <attribute name="title">
          <string>控制台</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QPlainTextEdit" name="consoleOutput">
            <property name="styleSheet">
             <string notr="true">background-color: rgb(31, 36, 48);
color: #55efc4;
font: 11pt Consolas;
border: none;
border-bottom: 1px solid #34495e;</string>
            </property>
            <property name="verticalScrollBarPolicy">
             <enum>Qt::ScrollBarAlwaysOff</enum>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="plainText">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <property name="spacing">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="label">
              <property name="styleSheet">
               <string notr="true">color: #00b894;</string>
              </property>
              <property name="text">
               <string>&gt;</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="consoleInput">
              <property name="styleSheet">
               <string notr="true">background-color: rgb(31, 36, 48);
color: #00b894;
margin: 0;
border: none;
font: 10pt &quot;Consolas&quot;;</string>
              </property>
              <property name="inputMask">
               <string/>
              </property>
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="memory">
         <property name="layoutDirection">
          <enum>Qt::LeftToRight</enum>
         </property>
         <property name="styleSheet">
          <string notr="true">background-color: rgb(31, 36, 48);</string>
         </property>
         <attribute name="title">
          <string>内存</string>
         </attribute>
         <layout class="QHBoxLayout" name="horizontalLayout_11" stretch="0">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_10" stretch="2,4">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_5">
              <property name="spacing">
               <number>0</number>
              </property>
              <item>
               <spacer name="verticalSpacer">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_7">
                <property name="spacing">
                 <number>0</number>
                </property>
                <item>
                 <widget class="QLineEdit" name="memPosition">
                  <property name="styleSheet">
                   <string notr="true">background-color: rgb(35, 40, 52);</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="label_4">
                  <property name="styleSheet">
                   <string notr="true">color: white;</string>
                  </property>
                  <property name="text">
                   <string>地址</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="memButton">
                  <property name="text">
                   <string>跳转</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_9">
                <property name="spacing">
                 <number>0</number>
                </property>
                <item>
                 <widget class="QLabel" name="label_5">
                  <property name="styleSheet">
                   <string notr="true">color: white;</string>
                  </property>
                  <property name="text">
                   <string>将</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="memAddress">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="minimumSize">
                   <size>
                    <width>30</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="styleSheet">
                   <string notr="true">background-color: rgb(35, 40, 52);
width: 50px;</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="label_6">
                  <property name="styleSheet">
                   <string notr="true">color: white;</string>
                  </property>
                  <property name="text">
                   <string>地址内容修改为</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="memValue">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="minimumSize">
                   <size>
                    <width>100</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="styleSheet">
                   <string notr="true">background-color: rgb(35, 40, 52);
width: 50px;</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="memchangeButton">
                  <property name="text">
                   <string>修改</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <spacer name="verticalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPlainTextEdit" name="memOutput">
              <property name="styleSheet">
               <string notr="true">border: none;
color: white;</string>
              </property>
              <property name="verticalScrollBarPolicy">
               <enum>Qt::ScrollBarAlwaysOff</enum>
              </property>
              <property name="horizontalScrollBarPolicy">
               <enum>Qt::ScrollBarAlwaysOff</enum>
              </property>
              <property name="undoRedoEnabled">
               <bool>false</bool>
              </property>
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab">
         <attribute name="title">
          <string>显存</string>
         </attribute>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <property name="spacing">
             <number>0</number>
            </property>
            <item>
             <widget class="QPlainTextEdit" name="vmOutput">
              <property name="styleSheet">
               <string notr="true">background-color: rgb(31, 36, 48);
color: #55efc4;
font: 11pt Consolas;
border: none;
border-bottom: 1px solid #34495e;</string>
              </property>
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_4">
              <property name="spacing">
               <number>0</number>
              </property>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
